### ISSUE TITLE
add `cml-send-comment --update`

### ISSUE THREAD

**Issue by casperdcl** [iterative/cml#512]
A new `--update` flag would try to overwrite an existing comment rather than post a new one.

**Comment by 0x2b3bfa0** [comment]
Would it update the latest comment for a given commit _ref_?

**Comment by casperdcl** [comment]
Would be maily for PRs, not for commits (it's quite rare to re-run commit workflows).

**Comment by 0x2b3bfa0** [comment]
Both `cml-pr` and `cml-send-comment` operate on comments: the former creates a pull request with an initial comment (description), while the latter adds new comments to a given commit. 🤔  Should we also repurpose the latter to update pull request descriptions?

**Comment by casperdcl** [comment]
I don't think we need to update PR descriptions.

**Comment by DavidGOrtega** [comment]
They work different, cml-pr creates a pr but does not send a SHA comment. Just only the PR description. And cml-pr is never going to reuse the pr. If you use it you will have a PR using current SHA, if happens that you are reusing SHA (repeat workflow) is returning the PR already created for that commit if exist or create if closed (treated as not existing)

**Comment by 0x2b3bfa0** [comment]
Thanks for the clarification, I got confused because of the new use case where `cml-pr` was ran directly from a branch.

**Comment by DavidGOrtega** [comment]
![image](https://user-images.githubusercontent.com/414967/118972043-65c7ca00-b970-11eb-90eb-2f6ec582dfba.png)

this could be very easy to be fixed.
If hash of current comment coincides with previous one skip


Another possibility is the use the watermark to even update with different changes ( that could happen through non deterministic ML models repeated CI execution not managed and tracked properly by DVC (dvc.lock)

**Comment by DavidGOrtega** [comment]
> I'd say that's a completely different feature which we need a separate issue

I would say is just one of the cases of what you want to solve. Indeed the CML watermark will help in both scenarios

**Comment by 0x2b3bfa0** [comment]
Blocked by [gitlab-org/gitlab#333000](https://gitlab.com/gitlab-org/gitlab/-/issues/333000)

**Comment by 0x2b3bfa0** [comment]
Related to #254, #285 and #294

**Comment by 0x2b3bfa0** [comment]
⚠️ This is **not** a substitute for proper end to end testing with ephemeral repositories.
